### PR TITLE
Add document for case sensitivity

### LINKED ISSUE #1657: How to create a DataFrame with column name case-sensitive
Hello, I find it is not easy to create a case-sensitive columns  DataFrame.
```py
data = {'a':[1,2], 'A': [2,3]}
kdf = ks.DataFrame(data)
...
pyspark.sql.utils.AnalysisException: Reference 'a' is ambiguous, could be: a, a.;
```
I googled this error and i should to set pyspark option `spark.sql.caseSensitive=true`.
So I tried `ks.set_option`, but it failed by `_check_option`.
Is there any way to set pyspark option? Or how to create a case-sensitive DataFrame?

### PR DESCRIPTION
**Pull request by shengjh**
Resolves #1657

### CODE REVIEW

File: docs/source/user_guide/best_practices.rst [hunk 1]
```diff
@@ -138,6 +138,17 @@ this behavior. For instance, see below:
...
+Additionally, it is strongly discouraged to use case sensitive column names. Koalas disallows it by default.
+
+.. code-block:: python
+
+   >>> import databricks.koalas as ks
+   >>> kdf = ks.DataFrame({'a': [1, 2], 'A':[3, 4]})
+   ...
+   Reference 'a' is ambiguous, could be: a, a.;
+
+However, you can turn on ``spark.sql.caseSensitive`` in spark configuration to enable it if you use on your own risk.
```
Comment (itholic) [inline docs/source/user_guide/best_practices.rst:150]: Could you add an example for this too ?

**Comment by itholic** [comment]
Otherwise, Seems good to me.

File: docs/source/user_guide/best_practices.rst (hunk 1 above)
Comment (shengjh) [reply docs/source/user_guide/best_practices.rst:150]: Done.  Please give it a look again.  :)

File: docs/source/user_guide/best_practices.rst [hunk 2]
```diff
@@ -138,6 +138,31 @@ this behavior. For instance, see below:
...
```
Comment (HyukjinKwon) [inline docs/source/user_guide/best_practices.rst:150]: Sorry, tiny nit: spark -> Spark

Comment (HyukjinKwon) [inline docs/source/user_guide/best_practices.rst:150]:
```suggestion
However, you can turn on ``spark.sql.caseSensitive`` in Spark configuration to enable it if you use on your own risk.
```

File: docs/source/user_guide/best_practices.rst (hunk 1 above)
Comment (itholic) [reply docs/source/user_guide/best_practices.rst]: Thanks, @shengjh !

**Pull request merged**